### ISSUE TITLE
[docs] Explain that coveralls "token" is NOT GITHUB_TOKEN

### ISSUE THREAD

**Issue by AlbinoGeek** [lemurheavy/coveralls-public#1508]
Attempting to setup coveralls for a Golang project using GitHub Actions resulted in:

```
Error: Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
```

The action showing this error can be seen here:
https://github.com/AlbinoGeek/sc2-rsu/runs/1545001536?check_suite_focus=true

The source for the action file can be seen here:
https://github.com/AlbinoGeek/sc2-rsu/blob/coveralls/.github/workflows/codeql-analysis.yml

This is the same symptom, but not solution as existing issues:

#632
#888
#984
#1268

**Comment by AlbinoGeek** [comment]
Fixed, the documentation should be updated however, to prevent this issue in the future.

When you setup a repository on Coveralls, it instructs you to add a GitHub Secret with your token.

However, clearly, it doesn't want this token, it actually wants your `GITHUB_TOKEN` -- which was misleading.

Creating a secret with the name `GITHUB_TOKEN` gives you the error "Invalid name" -- which is further misleading.

**Comment by afinetooth** [comment]
@AlbinoGeek I'd like to fix this, but can you tell me where you see in our docs that we instruct you to set an env var to your coveralls repo token? AFAIK, the [Coveralls Github Actions README](https://github.com/marketplace/actions/coveralls-github-action) is our only doc on Github Actions and it does not mention the repo token.

I do think an extra mention that the GITHUB_TOKEN is not the REPO_TOKEN would help, but that was the intent of [the Troubleshooting line here](https://github.com/marketplace/actions/coveralls-github-action#coveralls-responds-with-cannot-find-matching-repository):

> **Coveralls responds with "cannot find matching repository"**
> Ensure your workflow yaml line for the GitHub token matches exactly:
```
github-token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by AlbinoGeek** [comment]
The documentation I went from was that shown to me on the repository page (on Coveralls.io) after syncing, but before submitting a report. It was misleading that the page was telling me all about the "token" (referred to without specifying that this was a travis-specific token, or "non-GitHub" even) -- basically duplicate mention of the same terminology "token" -- without specifying which token was being discussed.

I would have figured that page would be specific to a repository synced from GitHub, because that's how I landed on the page. E.g: That it is showing me Travis instructions (or even non-github) instructions while I use GitHub was misleading. Perhaps, asking the user which CI they are using and only showing those instructions would help reduce this confusion.

**Comment by afinetooth** [comment]
Yes, I understand now. That start page is a legacy from when Travis was our only CI partner. The repo_token is mentioned there in the context of Travis, which can be read as "any CI" or "your CI"; but point taken, it would be better to clear that of specifics. 

In general, your experience indicates we can do better with our docs. Please know we're aware of that and are working on an update to be included in our next product update early next year.

**Comment by AlbinoGeek** [comment]
Thank you! Yes, this issue basically boils down to:

1) Start page only talks about Travis, and refers to `token` without specifying it's only for Travis.
1) Documentation for getting started with Golang did not mention having to use `cov 2 lcov` (separate issue).
1) A complete lack of "Getting started with Github Actions" documentation on Coveralls website itself.

**Comment by AlbinoGeek** [comment]
As mentioned elsewhere, this issue still exists and prevents new users from having a pleasant experience testing coveralls, which leads to a significantly higher bounce rate for these new users.

**Comment by afinetooth** [comment]
Hi, @AlbinoGeek. You're not wrong. We are close to releasing the documentation update for the coveralls.io service that includes a new version of the start page (the one you see after syncing a repo for the first time). 

It has taken longer than expected to complete this update, and that's not great. I'll post an update here when that's released.

**Comment by IT-VBFK** [comment]
Hi

I am struggling right now with this issue. (With GitHubActions)

I am using NUKE (a .NET build "engine") which uses coveralls.net. 

What I have tried so far:
- passing secrets.GITHUB_TOKEN to the build (not working, error like above)
- using a dedicated coveralls token (not working either, because github seems to prevent passing this secret to pipelines for PRs

Can you provide any help? Especially why GITHUB_TOKEN works on the coveralls github action, but not via coverall.net

**Comment by AlbinoGeek** [comment]
Stalebot, this issue still persists in published documentation.
